packages/lit-dev-content/src/components/algolia-search-controller.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {ArgsFunction, Task, TaskStatus} from '@lit-labs/task';
+import type {Hit} from '@algolia/client-search';
+export type {Hit} from '@algolia/client-search';
+import algoliasearch, {
+  SearchClient,
+  SearchIndex,
+} from 'algoliasearch/dist/algoliasearch-lite.esm.browser.js';
+import {ReactiveControllerHost} from 'lit';
+import {publicVars} from 'lit-dev-tools-esm/lib/configs.js';
+
+const agloliaSearchControllerDefaultOptions = {
+  appId: publicVars.algolia.appId,
+  searchOnlyKey: publicVars.algolia.searchOnlyKey,
+  index: publicVars.algolia.index,
+};
+
+export type AlgoliaSearchControllerOptions =
+  typeof agloliaSearchControllerDefaultOptions;
+
+export class AgloliaSearchController<T extends {}> {
+  host: ReactiveControllerHost;
+  private task;
+
+  private _client: SearchClient;
+  private _index: SearchIndex;
+  private _lastValue: Hit<T>[] = [];
+
+  public get value() {
+    if (!this.task || this.task.status !== TaskStatus.COMPLETE) {
+      return this._lastValue;
+    }
+
+    this._lastValue = this.task.value!;
+    return this.task.value!;
+  }
+
+  constructor(
+    host: ReactiveControllerHost,
+    argsFn: ArgsFunction<[string]>,
```

Comment:
You could type this as `() => string` so that the element doesn't have to put the query in an array.